README.md

```diff
@@ -70,13 +86,12 @@ To achieve this, the following approach is used.
 
 - Blanket implementations are provided for types that recursively contain types that implement the above traits eg. [`Option<T>`], [`Vec<T>`], [`Vec<Option<T>>`], [`Vec<Vec<Option<T>>>`], etc. The blanket implementation needs be enabled by the `arrow_enable_vec_for_type` macro on the primitive type. This explicit enabling is needed since Vec<u8> is a special type in Arrow, and implementation specialization is not yet supported in rust to allow blanket implementations to coexist with more specialized implementations.
 
-- Some additional supporting traits such as `ArrowMutableArrayTryPushGeneric` are used to workaround the unavailability of features like generic associated types.
+- Some additional supporting traits such are used to expose functionality that are not part of arrow2's public traits.
 
 ### Why not serde?
 
 While serde is the de-facto serialization framework in Rust, it introduces a layer of indirection.
-Specifically, arrow2 uses Apache Arrow's in-memory columnar format, while serde is row based.
-Using serde as an intermediary would lead to an extra allocation per non-primitive type per item, which is very expensive.
+Specifically, arrow2 uses Apache Arrow's in-memory columnar format, while serde is row based. Using serde requires a Serializer/Deserializer implementation around each arrow2 MutableArray and Array, leading to a heavyweight wrapper around simple array manipulations.
```

Comment:
Dug more into serde, and it's possible to implement via a memcopy. It just gets really messy since we'll need to implement all the serialize/deserialize methods (most of which remain unimplemented) around a simple array.